lib/ansible/module_utils/online.py

```diff
@@ -70,11 +70,21 @@ def get_resources(self):
         if not results.ok:
             raise OnlineException('Error fetching {0} ({1}) [{2}: {3}]'.format(
                 self.name, '%s/%s' % (self.module.params.get('api_url'), self.name),
-                results.status_code, results.json['message']
+                results.status_code, results.json['error']
             ))
 
         return results.json
 
+    def post_resources(self, data):
```

Comment:
Why not use a post method and have this treatment done elsewhere?

Reply:
I tried to use the same logic than you did with get and get_resources methods, am I wrong?